### ISSUE TITLE
Documentation doesn't state the AMP responders can return a Deferred

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4418]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#4418|
|Type|defect|
|Created|2010-04-29 09:40:09Z|
|Branch|https://github.com/twisted/twisted/tree/async-amp-responder-4418|

While AMP is stated as asynchronous, all the examples in the docstring or in the example directory are actually simple synchronous methods. It's not explicitly written anywhere that responders can actually return a Deferred, and everything will happen magically. I think the docstring (being the only real documentation) should be updated, and maybe the examples tweaked.

<details><summary>Searchable metadata</summary>

```
trac-id__4418 4418
type__defect defect
reporter__therve therve
priority__normal normal
milestone__ 
branch__branches_async_amp_responder_4418 branches/async-amp-responder-4418
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1272534009000000 1272534009000000
changetime__1273804857000000 1273804857000000
version__None None
owner__ 
cc__rwall
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @glyph|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [28889]) Branching to 'async-amp-responder-4418'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [28890]) fix it

refs [#4418](https://github.com/twisted/twisted/issues/4418)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_5"></a>|@wallrj set owner to @exarkun|
|-|-|

The new documentation makes perfect sense but I think it would be worth adding 
```
Sum.responder(slowSum)
```

...and perhaps adjust the new sentence and move the "transparent" comment up eg...
```
Command responders may also return Deferreds. The response will be 
sent when the Deferred fires. This is transparent to the caller.
```

But that's a matter of taste. 

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

> The new documentation makes perfect sense but I think it would be worth adding

Oh, yes, absolutely.  There's no way to defend the lack of that.  Added.

> perhaps adjust the new sentence and move the "transparent" comment up eg...

I didn't want to interrupt the flow if the explanation in the sentence leading up to the example.  I'm not terribly happy with the "transparent" sentence where it is, but I think I prefer it there to moving it up.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [28919]) Merge async-amp-responder-4418

Author: exarkun
Reviewer: rwall
Fixes: [#4418](https://github.com/twisted/twisted/issues/4418)

Add an example to the `twisted.protocols.amp` module docstring
of an AMP responder which returns a Deferred.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
